ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,475 @@
+    memory: "4 GB"
+      cpu: "1"
+        disks: "local-disk 100 HDD"
+  }
+}
+
+task MergeVCFs {
+  input {
+    Array[File] input_vcfs
+    String output_vcf_basename
+    Float shard_size
+  }
+  Int disk_size = ceil(((shard_size * 22) * 2) + 20)
+  command <<<
+    valid_vcfs=$(for file in ~{sep=' ' input_vcfs}; do rows=$(zgrep -v "#" $file | wc -l); if [ $rows -gt 0 ]; then echo $file; fi; done)
+    bcftools concat $valid_vcfs | bgzip -c > ~{output_vcf_basename}.vcf.gz
+  >>>
+  runtime {
+    docker: "farjoun/impute:0.0.3-1504715575"
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_vcf = "~{output_vcf_basename}.vcf.gz"
+  }
+}
+
+task UpdateHeader {
+  input {
+    File vcf
+    File ref_dict
+    Int disk_size = 400
+  }
+  command <<<
+  zgrep -v "##contig=<ID=" ~{vcf} | bgzip -c > no_contigs.vcf.gz 
+    gatk IndexFeatureFile -F no_contigs.vcf.gz
+    ## update the header of the merged vcf
+    gatk UpdateVCFSequenceDictionary --source-dictionary ~{ref_dict} --output final_call.vcf.gz --replace -V no_contigs.vcf.gz
+  >>>
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GiB"
+  }
+  output {
+    File output_vcf = "final_call.vcf.gz"
+    File output_vcf_index = "final_call.vcf.gz.tbi"   
+  }
+}
+
+task RemoveSymbolicAlleles {
+  input {
+    File original_vcf
+    File original_vcf_index 
+    String output_basename = "michigan_allchroms.no_symbolic"
+    Int disk_size = 200
+  }
+  command {
+    gatk SelectVariants -V ~{original_vcf} -xl-select-type SYMBOLIC --select-type-to-exclude MIXED --exclude-non-variants TRUE --remove-unused-alternates TRUE -O ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+   # docker: "us.gcr.io/broad-gatk/gatk:latest"
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GB"
+  }
+}
+
+task SeparateMultiallelics {
+  input {
+    File original_vcf
+    File original_vcf_index 
+    String output_basename = "michigan_allchroms.no_multi_symbolic"
+    Int disk_size = 200
+  }
+  command {
+    bcftools norm -m - ~{original_vcf} | bgzip -c > ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: "skwalker/imputation:with_python"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GB"
+  }
+}
+
+task SplitSample {
+  input {
+    File vcf
+    File vcf_index
+    String sample
+    Int disk_size = 400
+  }
+  command {
+    gatk SelectVariants -V ~{vcf} -sn ~{sample} -O ~{sample}.unsorted.vcf
+    gatk --java-options "-Xmx8g" SortVcf -I ~{sample}.unsorted.vcf -O ~{sample}.vcf.gz
```

Comment:
can these be piped together to save disk space, and probably also speed?

Reply:
It shouldn't be necessary to do both ( I think this was some issue I was having in like January with gzipped vs normal vcfs but I think you might have actually fixed this ?)

Reply:
ohhh right I remember, it was this bug: broadinstitute/gatk#6443

that is (for our purposes) fixed in gatk now, so SelectVariants can output a gzipped vcf, and the output will be correctly sorted so you don't need the call to `SortVcf`